includes/class-sensei-usage-tracking.php

```diff
@@ -0,0 +1,321 @@
+		}
+	}
+
+	public static function maybe_unschedule_tracking_task() {
+		if ( wp_next_scheduled( self::$job_name ) ) {
+			wp_clear_scheduled_hook( self::$job_name );
+		}
+	}
+
+	/**
+	 * Check if tracking is enabled.
+	 * @return bool
+	 **/
+	public static function is_tracking_enabled() {
+		$setting = Sensei()->settings->get( self::$usage_tracking_setting_name ) || false;
+		/**
+		 * Filter
+		 * @return bool
+		 **/
+		$enabled = (bool) apply_filters( 'sensei_usage_tracking_enabled', (bool)$setting );
+
+		return $enabled;
+	}
+
+	/**
+	 * Attach hooks.
+	 **/
+	function hook() {
+		// Setting
+		add_filter( 'sensei_settings_fields', array( $this, 'add_setting_field' ) );
+		// Admin
+		add_action( 'admin_notices', array( $this, 'maybe_display_tracking_opt_in' ) );
+		add_action( 'admin_menu', array( $this, 'register_usage_submenu_page' ) );
+		add_action( 'admin_init', array( $this, 'handle_request' ) );
+		// Cron
+		add_filter( 'cron_schedules', array( $this, 'add_two_weeks' ) );
+		add_action( self::$job_name, array( $this, 'maybe_send_usage_data' ) );
+	}
+
+	function add_two_weeks( $schedules ) {
+		$schedules['sensei_two_weeks'] = array(
+			'interval' => 15 * DAY_IN_SECONDS,
+			'display'  => esc_html__( 'Every Two Weeks', 'woothemes-sensei' ),
+		);
+
+		return $schedules;
+	}
+
+	function handle_request() {
+		if ( ! isset( $_GET['page'] ) ) {
+			return;
+		}
+
+		if ( $_GET['page'] !== 'sensei_usage_tracking' ) {
+			return;
+		}
+
+		if ( isset( $_POST['usage_tracking_action'] ) && $_POST['usage_tracking_action'] === 'send_data' ) {
+			check_admin_referer( 'send_data_nonce' );
+			$result = self::maybe_send_usage_data();
+
+			if ( is_wp_error( $result ) ) {
+				wp_redirect( 'admin.php?page=sensei_usage_tracking&msg=data_send_error' );
+			}
+
+			wp_redirect( 'admin.php?page=sensei_usage_tracking&msg=data_send_success' );
+			exit;
+		}
+	}
+
+	/**
+	 * Send usage data.
+	 **/
+	public static function maybe_send_usage_data() {
+		if ( ! self::is_tracking_enabled() ) {
+			return;
+		}
+
+		/**
+		 * Define data to send. Add or remove array keys and values.
+		 *
+		 * @param array $usage_data The data to send.
+		 *
+		 * @return array The array should be key/value. All values will be urlencoded
+		 **/
+		$usage_data = (array) apply_filters( 'sensei_usage_tracking_usage_data', array(
+			'course_count' => wp_count_posts( 'course' )->publish,
+			'learner_count' => self::get_learner_count(),
+			'lesson_count' => wp_count_posts( 'lesson' )->publish,
+			'message_count' => wp_count_posts( 'sensei_message' )->publish,
+			'question_count' => wp_count_posts( 'question' )->publish,
+			'teacher_count' => self::get_teacher_count(),
+		) );
+
+		$resp = self::send_event( 'stats_log', $usage_data );
+
+		// Send a dummy event at the same time to enable using Tracks funnels,
+		// which requires a minimum of two event steps.
+		$resp = self::send_event( 'sensei_dummy_stats_log' );
```

Comment:
I actually don't care about conversions. 😄 It's not why I'm using funnels. I'm using them to enable targeted messaging. I want to answer the question "What is the admin email address of sites that have created 10+ courses?"

By way of an example, I can create a funnel where the first step is set to `course_count = 10` (I'd really like to be able to do >=10, but that's not possible I don't think). Then if I create a cohort from that first step I'll be able to see the site URL (stored in `userid`) of the sites that have created 10 courses. The data returned by the cohort doesn't show our custom `admin_email` event property though, so I'll likely have to use the site URL and cross-reference with purchaser info in woocommere.com. I can then email the customer and ask "Hey, we see you've created a lot of courses. What were some of your pain points?"

Showing the event prop values in the cohort data would make my life a lot easier though, as would being able to specify comparison operators like >, <, = in funnels. I'm having to find some clever (sneaky?) ways of getting the data I need from Tracks. 🙂

Reply:
I see, ok so for what you are currently doing you can just build a cohort directly without going through the funnel rigamarole. We allow cohorts of users to be built directly with the cohorts tool:

https://mc.a8c.com/tracks/cohorts/new/

So in this case instead of a funnel you can just define a cohort as anyone (a.k.a. any site in this case) that has performed the `sensei_stats_log` event with `course_count = 10`.

Having a feature to do inequalities when searching properties themselves is an interesting idea, maybe something we can do in the future but in the mean time we can also give you a query to get these users outside of the cohorts tool. (We have this thing called Hue that can be used to run arbitrary queries and export data.)

Anyway let's get rid of the dummy event as it's not really needed in any event.

Reply:
Removed in https://github.com/Automattic/sensei/pull/1965/commits/d0c5922b6cfb7b42635e129e6aaa65ad7cee2bc8. Thx!